### ISSUE TITLE
[glib] Build error on x64-windows

### ISSUE THREAD

**Issue by YunchengLiu** [microsoft/vcpkg#35980]
on windows11 23H2 with MSVC version 17.8.3
command：.\vcpkg.exe install glib:x64-windows
log：
[package-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/13807727/package-x64-windows-dbg-out.log)

**Comment by JonLiu1993** [comment]
@YunchengLiu, Thanks for posting this issue, which python version are you using?
````
ImportError: DLL load failed while importing pyexpat: 找不到指定的模块。
````

**Comment by YunchengLiu** [comment]
All installed python instances on my system are managed through Miniconda3, and I haven't set any environment variables for system-range access.

Is it related to the embedded Python version (3.11.5) that comes with VCPKG?

**Comment by JonLiu1993** [comment]
I cannot reproduce your error locally. This error is related to your python version. It is also possible that the Python you installed is probably missing expat, which is why you cannot import it.

**Comment by YunchengLiu** [comment]
I'm very sorry for the late reply.

In my system, there are no globally accessible Python instances without entering Conda.

I also tried setting up an additional global Python interpreter (3.12.1 through Scoop), but the problem still occurred.

Does the content of the log seem to be related to accessing embedded Python downloaded from VCPKG?

Or do I need other operations? I'm a bit confused now.

**Comment by JonLiu1993** [comment]
Could you clone a refresh vcpkg and try again?

**Comment by YunchengLiu** [comment]
I cloned a VCPKG instance again and then executed ` \vcpkg.exe installed glib: x64-windows ` and received the same error. The log looks the same.
[package-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/13973112/package-x64-windows-dbg-out.log)

**Comment by YunchengLiu** [comment]
I have added a new pth file to the vcpkg's python directory (`.../vcpkg/downloads/tools/Python/python-3.11.5-x64/python311. _pth`).

This problem has been solved at least in my environment.

The content of the pth file is as follows:
```Plain text
./
./python311.zip
```

It seems that some Python scripts called during the glib installation process have changed the library search behavior of this Python environment, or perhaps for other reasons.(I'm not familiar with glib, especially what its installation process is doing with python)

Since I do not currently have any other development environment, I am not sure if this is a universal solution to the problem.
